kvbc/include/categorization/block_merkle_category.h

```diff
@@ -111,6 +111,13 @@ class BlockMerkleCategory {
   sparse_merkle::Tree tree_;
 };
 
+inline const MerkleValue& asMerkle(const Value& v) { return std::get<MerkleValue>(v); }
+inline MerkleValue& asMerkle(Value& v) { return std::get<MerkleValue>(v); }
+
+// Optional overloads assume the optional contains a value.
+inline const MerkleValue& asMerkle(const std::optional<Value>& v) { return asMerkle(*v); }
```

Comment:
As @dartdart26 pointed out in another comment, this is incredibly unsafe. IMO We shouldn't wrap unrestricted calls that assume an optional contains a value. The use of the `*` operator is best left to only being used immediately after a check that the optional contains a value, in inline code. Otherwise, any modification of the code can break the implicit assumption since the user may not look up the `asMerkle` function and notice that it uses `*`. 

I also personally, am not a fan of wrapping `std::get`, but at least that code will throw an exception and not lead to silent corruption.

Reply:
I copy-pasted it from the other categories.
I removed the functions the accept optional, but for the other categories, I guess it should be done as well.